### DISCUSSION TITLE
Labels outside and invisible at pie chart

### DISCUSSION THREAD

**Discussion by konrad79** [chartjs/chartjs-plugin-datalabels#237] [Q&A] [answered]
I want to add labels around a pie chart (in Angular 11). But since the circle of the chart is as large that it reaches the "bottom-line" of the canvas, the bottom labels are not visible.
Is there a setting that calculate other, visible positions for the labels? Or how can I shrink the size of the pie to get enough space for the bottom labels?

![2021-04-22 15_54_13-StatistikApp - Übersicht der SelectLine Lizenzstatistiken und Kundenstatistiken](https://user-images.githubusercontent.com/31411681/115726453-07b0c280-a383-11eb-87ef-cd1377ff6b44.png)

Typescript code:
```js
plugins: {
  datalabels: {
    display: this.showLegend,
    align: 'end',
    anchor: 'end',
    clamp: true,
    font: { size: 12, weight: 'normal' },
    color: '#333',
    formatter(value, context) {
      return context.chart.data.labels[context.dataIndex];
    },
  }
}
```

**Comment by kurkle** [comment, marked as answer]
You can use layout padding to make room for the labels:
https://www.chartjs.org/docs/latest/configuration/layout.html

**Reply by konrad79** [reply]
Thx :-)

**Comment by Rjaat** [comment]
> I want to add labels around a pie chart (in Angular 11). But since the circle of the chart is as large that it reaches the "bottom-line" of the canvas, the bottom labels are not visible. Is there a setting that calculate other, visible positions for the labels? Or how can I shrink the size of the pie to get enough space for the bottom labels?
> 
> ![2021-04-22 15_54_13-StatistikApp - Übersicht der SelectLine Lizenzstatistiken und Kundenstatistiken](https://user-images.githubusercontent.com/31411681/115726453-07b0c280-a383-11eb-87ef-cd1377ff6b44.png)
> 
> Typescript code:
> 
> ```js
> plugins: {
>   datalabels: {
>     display: this.showLegend,
>     align: 'end',
>     anchor: 'end',
>     clamp: true,
>     font: { size: 12, weight: 'normal' },
>     color: '#333',
>     formatter(value, context) {
>       return context.chart.data.labels[context.dataIndex];
>     },
>   }
> }
> ```

i am not even able to add labels around the pie-chart. can you help with this?? (using chartjs-plugin-datalabels version 2.2.0 and Chart.js v4.4.1))

**Comment by Rohitpatil-123** [comment]
![image](https://github.com/user-attachments/assets/78ebe636-003d-45e1-83a5-d84c186bef84)
these labels are overlapping with each other and also want to increase the width of bar could you please help me @kurkle @konrad79

**Reply by konrad79** [reply]
@Rohitpatil-123 What are you trying to achieve? More space between the bars? Auto cutting to long labels with "..."? Smaller fonts for the labels?
